src/osgPlugins/dae/daeRGeometry.cpp

```diff
@@ -160,18 +152,15 @@ osg::Node* daeReader::processMorph(domMorph* pDomMorph, domBind_material* pDomBi
     }
 
     // Base mesh
-    osg::Geode* pOsgGeode = getOrCreateGeometry(pDomGeometry, pDomBindMaterial);
-    if (!pOsgGeode)
+    osg::Group* pOsgGeometryGroup = getOrCreateGeometry(pDomGeometry, pDomBindMaterial);
+    if (!pOsgGeometryGroup)
         return NULL;
 
-    // Expects a single geometry inside the geode, should change this
-    osg::Geometry* pOsgGeometry = dynamic_cast<osg::Geometry*>(pOsgGeode->getDrawable(0));
-    if (!pOsgGeometry)
-        return NULL;
+    // Expects a single geometry inside the group, should change this
+    osg::Geometry* pOsgGeometry = dynamic_cast<osg::Geometry*>(pOsgGeometryGroup->getChild(0));
 
     osgAnimation::MorphGeometry* pOsgMorphGeometry = new osgAnimation::MorphGeometry(*pOsgGeometry);
-    pOsgGeode->removeDrawables(0);
-    pOsgGeode->addDrawable(pOsgMorphGeometry);
+    pOsgGeometry = pOsgMorphGeometry;
```

Comment:
Is this method safe? Is there a chance of memory leak? This creates osgAnimation::MorphGeometry(*pOsgGeometry), then changes the pOsgGeometry pointer to point to the new object. I haven't checked what osgAnimation::MorphGeometry constructor actually does.